packages/commerce-sdk-react/src/hooks/ShopperBaskets/helpers.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useCustomerId, useShopperBasketsMutation} from '../index'
+import {useCustomerBaskets} from '../ShopperCustomers'
+import {onClient} from '../../utils'
+import {ApiClients, Argument} from '../types'
+type Client = ApiClients['shopperBaskets']
+
+export function useShopperBasketsMutationHelper() {
+    const customerId = useCustomerId()
+    const {data: basketsData} = useCustomerBaskets(
+        {parameters: {customerId}},
+        {
+            enabled: !!customerId && onClient()
+        }
+    )
+    const createBasket = useShopperBasketsMutation('createBasket')
+    const addItemToBasketMutation = useShopperBasketsMutation('addItemToBasket')
+    return {
+        // a function is responsible for managing the process of adding an item to a basket.
+        // If a basket already exists, add the item to the basket immediately.
+        // If a basket does not exist, create a new basket using the createBasket mutation
+        // and then add the item to the newly created basket using the addItemToBasket mutation.
+        addItemToNewOrExistingBasket: async (
+            body: Argument<Client['addItemToBasket']> extends {body: infer B} ? B : undefined
```

Comment:
I'm wondering if we should be more opinionated about the argument name here. When we implemented the client function for the API endpoint we followed the schema which is the right thing to do. But we aren't following the schema completely here since we don't have a params argument. Would it be easier to read if it was called `productItems` over `body`?

Reply:
Also, are we defining a return type for this helper method?

Reply:
I've made the change for `body` and added return type for `addItemToNewOrExistingBasket`